wsl-builder/release-info/release-info.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/ubuntu/wsl/wsl-builder/common"
+)
+
+type columns struct {
+	rootfsArch                                         *string
+	rootfs, appId, fullName, launcher, codeName, short *bool
+}
+
+func writeReleaseInfo(csvPath string, distros []string, cols columns) (exitCode int) {
+	exitCode = 0
+
+	releases, ok, err := selectReleases(csvPath, distros)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		return 1
+	}
+	if !ok {
+		exitCode = 2
+	}
+
+	if ok := writeTable(releases, cols); !ok {
+		exitCode = 2
+	}
+
+	return exitCode
+}
+
+// writeTable writes the information in a table into stdout.
+// Issues are reported to stderr.
+// Returns OK if there were no issues.
+func writeTable(releases []common.WslReleaseInfo, cols columns) (ok bool) {
+	w := tabwriter.NewWriter(os.Stdout, 1, 2, 2, ' ', 0)
+	defer w.Flush()
+
+	ok = true
+	if e := writeHeader(w, cols); e != nil {
+		ok = false
+		fmt.Fprintf(os.Stderr, "Warning: could not write header: %v\n", e)
+	}
+
+	for _, r := range releases {
+		if e := writeRow(w, r, cols); e != nil {
+			ok = false
+			fmt.Fprintf(os.Stderr, "Warning: could not write row for release %q: %v\n", r.AppID, e)
```

Comment:
ditto on decorating the error in the child function. Then, all of this function would be way more compact:

```go
func writeTable(releases []common.WslReleaseInfo, cols columns) (err error) {
	w := tabwriter.NewWriter(os.Stdout, 1, 2, 2, ' ', 0)
	defer w.Flush()

	if err := writeHeader(w, cols); err != nil {
		return err	
	}

	for _, r := range releases {
		if err := writeRow(w, r, cols); err != nil {
			return fmt.Errorf("got an error for release %q: %v", r.AppID, err)
		}
	}

	return nil
}
```